hudi-common/src/main/java/org/apache/hudi/common/util/SchemaCompatibility.java

```diff
@@ -0,0 +1,566 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.util;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.avro.AvroRuntimeException;
+import org.apache.avro.Schema;
+import org.apache.avro.Schema.Field;
+import org.apache.avro.Schema.Type;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * NOTE: This code is copied from org.apache.avro.SchemaCompatibility and changed for HUDI use case.
```

Comment:
this needs to be attributed in LICENSE and NOTICE ..  Can we document all the changes? I am wondering if we can just wrap or extend the `org.apache.avro.SchemaCompatibility` class instead of copying in full.. 


cc @bvaradar as well..

Reply:
The functions that are changed in org.apache.avro.SchemaCompatibility are all private.

I can reduce the size of this file by referring to classes from org.apache.avro.SchemaCompatibility but I think starting with a copy makes it easier to see a diff.

Reply:
Not required any longer as I have removed the copied file. Please see the latest diff.